### PR TITLE
Update Kron const diag

### PR DESCRIPTION
**Pull request by wjmaddox**
Updates the solves and root decompositions when `D = \kron a_i I`.

### CODE REVIEW

File: gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py [hunk 1]
```diff
@@ -144,44 +159,41 @@ def _solve(self, rhs, preconditioner=None, num_tridiag=0):
...
+            res1 = evecs._transpose_nonbatch().matmul(dlt_inv_root.matmul(rhs))
+            res2 = evals_p_i.inv_matmul(res1)
+            res3 = evecs.matmul(res2)
+            res = dlt_inv_root.matmul(res3)
+            return res.to(rhs_dtype)
+
+            tmp_term = evecs.matmul(evals_p_i.inv_matmul(evecs._transpose_nonbatch().matmul(rhs)))
             res = lt._inv_matmul(rhs - tmp_term)
 
             return res.to(rhs_dtype)
```
Comment (Balandat) [inline gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py:199]: something got mixed up here...

**Review by Balandat — CHANGES_REQUESTED** [review]

File: gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py (hunk 1 above)
Comment (wjmaddox) [reply gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py:199]: I'm not following. I renamed the the `S` `Lambda_I` throughout to be more consistent.

Comment (wjmaddox) [reply gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py:199]: Oh nevermind i see it.

Comment (Balandat) [reply gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py:199]: you have two returns in the same indent block...

**Review by Balandat — APPROVED** [review]

**Pull request merged**